internal/resource/mutation/mutation.go

```diff
@@ -69,6 +69,23 @@ func (o *Op) Apply(ctx context.Context, comp *apiv1.Composition, current, mutate
 	return Apply(o.Path, mutated.Object, o.Value)
 }
 
+// unquoteKey removes quotes from a key string, handling both single and double quotes
+func unquoteKey(key string) string {
+	if len(key) >= 2 {
+		if (key[0] == '"' && key[len(key)-1] == '"') || (key[0] == '\'' && key[len(key)-1] == '\'') {
```

Comment:
Maybe a regex would be simpler? And it could cover both the single and double quote case